lib/data/repositories/v2/purchase_repository.dart

```diff
@@ -16,28 +16,30 @@ class PurchaseRepository {
 
   /// Initiate a new Purchase Request. The return is a purchase request
   /// with payment details on how to pay for the purchase
-  Future<Either<RequestFailure, InitiatePurchase>> initiatePurchase(
+  Future<Either<ServerFailure, InitiatePurchase>> initiatePurchase(
     int productId,
     PaymentType paymentType,
   ) async {
-    return executor.execute(
+    final result = await executor(
       () => apiV2.apiV2PurchasesPost(
         body: InitiatePurchaseRequest(
           productId: productId,
           paymentType: paymentTypeToJson(paymentType),
         ),
       ),
-      (dto) => InitiatePurchase.fromDto(dto),
     );
+
+    return result.bind((result) => Right(InitiatePurchase.fromDto(result)));
```

Comment:
Use map instead of bind
```suggestion
    return result.map(InitiatePurchase.fromDto);
```